modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryServiceTest.java

```diff
@@ -640,6 +655,96 @@ public void testSearchObjectEntries() throws Exception {
 		_objectEntryLocalService.deleteObjectEntry(objectEntry2);
 	}
 
+	@Test
+	public void testValidateMaximumNumberOfObjectEntries() throws Exception {
+		_setUser(_guestUser);
+
+		HashMapDictionaryBuilder.HashMapDictionaryWrapper<String, Object>
+			objectConfigurationDictionary = HashMapDictionaryBuilder.create(
+				Collections.emptyMap());
+
+		HashMapDictionaryBuilder.put(
+			"maximumNumberOfGuestUserObjectEntriesPerObjectDefinition", 1);
+		HashMapDictionaryBuilder.put("duration", 1);
+		HashMapDictionaryBuilder.put("timeScale", "days");
+		HashMapDictionaryBuilder.put("maximumFileSizeForGuestUsers", 25);
+
+		ConfigurationTestUtil.saveConfiguration(
+			ObjectConfiguration.class.getName(),
+			objectConfigurationDictionary.build());
+
+		Role guestRole = _roleLocalService.getRole(
+			TestPropsValues.getCompanyId(), RoleConstants.GUEST);
+
+		_resourcePermissionLocalService.addResourcePermission(
+			TestPropsValues.getCompanyId(), _objectDefinition.getResourceName(),
+			ResourceConstants.SCOPE_COMPANY,
+			String.valueOf(TestPropsValues.getCompanyId()),
+			guestRole.getRoleId(), ObjectActionKeys.ADD_OBJECT_ENTRY);
+
+		Assert.assertNotNull(
+			_objectEntryService.addObjectEntry(
+				0, _objectDefinition.getObjectDefinitionId(),
+				Collections.emptyMap(),
+				ServiceContextTestUtil.getServiceContext(
+					TestPropsValues.getGroupId(), _guestUser.getUserId())));
+
+		AssertUtils.assertFailure(
+			ObjectEntryCountException.class,
+			StringBundler.concat(
+				"Unable to exceed ", 1,
+				" guest object entries for object definition ",
+				_objectDefinition.getObjectDefinitionId()),
+			() -> _objectEntryService.addObjectEntry(
+				0, _objectDefinition.getObjectDefinitionId(),
+				Collections.emptyMap(),
+				ServiceContextTestUtil.getServiceContext(
+					TestPropsValues.getGroupId(), _guestUser.getUserId())));
+
+		ConfigurationTestUtil.deleteConfiguration(
+			ObjectConfiguration.class.getName());
+
+		HashMapDictionaryBuilder.put(
+			"maximumNumberOfGuestUserObjectEntriesPerObjectDefinition", 2);
+		HashMapDictionaryBuilder.put("timeScale", "weeks");
+
+		ConfigurationTestUtil.saveConfiguration(
+			ObjectConfiguration.class.getName(),
+			objectConfigurationDictionary.build());
+
+		LocalDate beginningDate = LocalDate.now(
```

Comment:
I think you can remove the minusDays(7) from this line and rename the variable to localDate, then you apply the minusDays(7) in the next line. Is this possible?